packages/aws-rfdk/lib/lambdas/nodejs/lib/configure-spot-event-plugin/spot-event-plugin-client.ts

```diff
@@ -94,6 +118,47 @@ export class SpotEventPluginClient {
     }
   }
 
+  public async getCollection(type: CollectionType): Promise<string[]> {
+    console.log(`Getting ${type.toString()} collection:`);
+    try {
+      const response = await this.deadlineClient.GetRequest(`/db/settings/collections/${type.toString()}s?invalidateCache=true`, {
+        headers: {
+          'Content-Type': 'application/json; charset=utf-8',
+        },
+      });
+      const deadlinePools: PoolCollections = response.data;
+      if (!deadlinePools.Pools || !Array.isArray(deadlinePools.Pools)) {
+        console.error(`Failed to receive a ${type.toString()} collection. Invalid response: ${deadlinePools}.`);
+        return [];
+      }
+      return deadlinePools.Pools;
+    } catch(e) {
+      console.error(`Failed to get ${type.toString()} collection. Reason: ${e}`);
+      return [];
+    }
+  }
+
+  public async saveCollection(pools: string[], type: CollectionType): Promise<boolean> {
+    console.log(`Saving ${type.toString()} collection:`);
+    console.log(pools);
+
+    try {
+      await this.deadlineClient.PostRequest(`/db/settings/collections/${type.toString()}s/save`, {
+        Pools: pools,
+        ObsoletePools: [],
```

Comment:
Just to make sure, would this overwrite what the old values of `ObsoletePools` are in Deadline with an empty array? or does it retain the old value for `ObsoletePools`?

Reply:
Good catch. I have check and `ObsoletePools` is rewritten with empty values. Fixed.